pkg/collectors/node.go

```diff
@@ -252,19 +277,74 @@ func (nc *nodeCollector) collectNode(ch chan<- prometheus.Metric, n v1.Node) {
 		addGauge(descNodeStatusPhase, boolFloat64(p == v1.NodeTerminated), string(v1.NodeTerminated))
 	}
 
-	// Add capacity and allocatable resources if they are set.
-	addResource := func(d *prometheus.Desc, res v1.ResourceList, n v1.ResourceName) {
-		if v, ok := res[n]; ok {
-			addGauge(d, float64(v.MilliValue())/1000)
+	if !nc.opts.DisableNodeNonGenericResourceMetrics {
+		// Add capacity and allocatable resources if they are set.
+		addResource := func(d *prometheus.Desc, res v1.ResourceList, n v1.ResourceName) {
+			if v, ok := res[n]; ok {
+				addGauge(d, float64(v.MilliValue())/1000)
+			}
+		}
+
+		addResource(descNodeStatusCapacityCPU, n.Status.Capacity, v1.ResourceCPU)
+		addResource(descNodeStatusCapacityNvidiaGPU, n.Status.Capacity, v1.ResourceNvidiaGPU)
+		addResource(descNodeStatusCapacityMemory, n.Status.Capacity, v1.ResourceMemory)
+		addResource(descNodeStatusCapacityPods, n.Status.Capacity, v1.ResourcePods)
+
+		addResource(descNodeStatusAllocatableCPU, n.Status.Allocatable, v1.ResourceCPU)
+		addResource(descNodeStatusAllocatableNvidiaGPU, n.Status.Allocatable, v1.ResourceNvidiaGPU)
+		addResource(descNodeStatusAllocatableMemory, n.Status.Allocatable, v1.ResourceMemory)
+		addResource(descNodeStatusAllocatablePods, n.Status.Allocatable, v1.ResourcePods)
+	}
+
+	capacity := n.Status.Capacity
+	allocatable := n.Status.Allocatable
+
+	for resourceName, val := range capacity {
+		switch resourceName {
+		case v1.ResourceCPU:
+			addGauge(descNodeStatusCapacity, float64(val.MilliValue())/1000, sanitizeLabelName(string(resourceName)), string(constant.UnitCore))
+		case v1.ResourceStorage:
+			fallthrough
+		case v1.ResourceEphemeralStorage:
+			fallthrough
+		case v1.ResourceMemory:
+			addGauge(descNodeStatusCapacity, float64(val.MilliValue())/1000, sanitizeLabelName(string(resourceName)), string(constant.UnitByte))
+		case v1.ResourceNvidiaGPU:
```

Comment:
is there a tracking issue to remove this resource in a follow-on?

Reply:
There is no tracking issue. But we have make a agreement that once Kubernetes 1.11 is released it will be removed in #417